src/nvim/screen.c

```diff
@@ -1570,6 +1570,17 @@ static void win_update(win_T *wp)
     got_int = save_got_int;
 }
 
+///
+/// Returns maximum length of the signcolumn
+///
+/// @param wp window we want signcolumn width from
+///
+/// @return max width of signcolumn (cell unit)
+int win_get_colwidth(win_T *wp)
+{
+  // 2 is the default value
+  return 2;
```

Comment:
No objection to this direction, just let's be sure to name it and document it so that things are less confusing than before :) 

It sounds like `buf_signcol_width()` would make the most sense. Windows don't have signs, buffers do (and that's hinted at by the existing internal functions).

> I read several times on the gitter it was a concern that neomake signs override vim-signature marks for instance; it bothers me as well

Ok, but we'd still want some sort of limit I imagine.